lib/browser/api/web-contents.js

```diff
@@ -120,7 +130,18 @@ const asyncWebFrameMethods = function (requestId, method, callback, ...args) {
         if (typeof callback === 'function') callback(result)
         resolve(result)
       } else {
-        reject(error)
+        if (error && error.__ELECTRON_SERIALIZED_ERROR__) {
+          let rehydratedError = error
+
+          if (errorConstructors[error.name]) {
```

Comment:
I guess this check can be moved to a condition above, so the whole change will be:

```js
} else {
  if (/* is a serialized error */) {
    const rehydratedError = new errorConstructors[error.name](error.message)
    rehydratedError.stack = error.stack
    reject(rehydratedError)
  } else {
    reject(error)
  }
}
```